src/components/extractionQc/ExtractionQcForm.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div>
+    <traction-form class="text-left" @submit="onSubmit">
+      <traction-sub-section title="Which QC Results would you like to upload?" class="py-6">
+        <traction-select
+          v-model="usedBySelected"
+          :options="usedByOptions"
+          :state="Boolean(usedBySelected) ? true : null"
+          required
+        ></traction-select>
+      </traction-sub-section>
+      <traction-sub-section title="CSV File" class="py-6">
+        <traction-file
+          v-model="file"
+          :state="Boolean(file) ? true : null"
+          placeholder="Choose a file or drop it here..."
+          drop-placeholder="Drop file here (CSV only)..."
+          accept="text/csv, .csv"
+          required
+        ></traction-file>
+      </traction-sub-section>
+      <traction-button id="upload-button" type="submit" theme="create" size="lg" :disabled="busy">
+        <!-- Weird bug - spinner won't show unless button text is two words/ big enough? -->
+        Upload File
+        <traction-spinner v-show="busy"></traction-spinner>
+      </traction-button>
+    </traction-form>
+  </div>
+</template>
+
+<script>
+import Api from '@/mixins/Api'
+import { createQcResultsUploadResource } from '@/services/traction/QcResultsUpload'
+
+const EXTRACTION_USED_BY = 'extraction'
+
+export default {
+  name: 'ExtractionQcForm',
+  mixins: [Api],
+  data() {
+    return {
+      usedByOptions: [
+        { value: null, text: 'Please select a option' },
+        { value: EXTRACTION_USED_BY, text: 'Extraction' },
+      ],
+      file: null,
+      usedBySelected: null,
+      busy: null,
+    }
+  },
+  computed: {
+    qcResultUploadsRequest: ({ api }) => api.traction.qc_results_uploads.create,
```

Comment:
does this need to be done here and if so does it need to be computed? could it just be added directly into the QcResultsUpload file?

Reply:
@BenTopping @stevieing I think it would defined in this file, unless I pass in the store object to the `createQcResultsUploadResource` function, so not sure that helps it much?
I could remove it from a computed, but the api function is defined as a computed prop in the Api module, so thought this was neater than `let qcResultUploadsRequest = this.$store.getters.api.traction.qc_results_uploads.create`. But don't mind

Reply:
I've left as is for now

Reply:
OK. Generally computed properties re-render quite a few times which would be unnecessary in this case but should be fine to leave in